test/unit/parse.test.ts

```diff
@@ -263,10 +263,24 @@ describe('parse', function() {
     })
   })
 
+  it('.parse standalone match criteria', function() {
+    const config = parse(`
+      Match all canonical final
+    `)
+    const match = config.find(line => line.type === DIRECTIVE && line.param === 'Match')
+    assert.ok(match)
+    assert.ok('criteria' in match)
+    assert.deepEqual(match.criteria, {
+      all: [],
+      canonical: [],
+      final: [],
+    })
+  })
+
   // https://github.com/cyjake/ssh-config/issues/58
   it('.parse match criteria', function() {
     const config = parse(`
-      Match exec "/Users/me/onsubnet --not 192.168.1." host docker
+      Match exec "/Users/me/onsubnet --not 192.168.1." final host docker
```

Comment:
is the `final` token required per standard?

Reply:
It's one of the three standalone criteria. I can't find mention of it in some manpages, but it is documented here https://man7.org/linux/man-pages/man5/ssh_config.5.html

Adding it here is just to test that it gets parsed without treating the next thing as an argument

Reply:
got it